### PR TITLE
ci: tag ceph image with quay.io

### PR DESCRIPTION
**Pull request by Madhu-1**
added if-else check to check the image starting with quay.io. If yes, pull and tag the image with quay.io

i cannot reopen https://github.com/ceph/ceph-csi/pull/2420, this is a replacement PR.

updates #2797

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@nixpanic to answer your  questions at  #2420

>Might be easier to use short names (without quay.io or docker.io) in build.env and other placed?

If we do this the image will get pulled from the default docker hub registry?

>Things might be easier, and only a change to the images.txt could work.

as the retagging is happed with docker.io in CI jobs we might not able able to find image with quay.io and re-pull happens and it takes times

File: mirror/images.txt [hunk 1]
```diff
@@ -6,6 +6,9 @@
 # format: <full-image-spec> <short-name>
 #
 
+# ceph-csi devel
```
Comment (nixpanic) [inline mirror/images.txt:9]: this is not needed, quay.io does not have pull limits like docker

remove the docker.io/ceph/ceph entry below.

**Review by nixpanic — CHANGES_REQUESTED** [review]
I think this does more than needed. When using images from quay. there is no need to mirror them. Use `quay.io/ceph/ceph:v16` everywhere, instead of `ceph/ceph:v16`.

How does Rook refer to the images? If that is the abbreviated name, only then you may need to do the re-tagging with `podman_pull`.

**Comment by Madhu-1** [comment]
> I think this does more than needed. When using images from the quay. there is no need to mirror them. Use `quay.io/ceph/ceph:v16` everywhere, instead of `ceph/ceph:v16`.
> 

Yes, it can be removed as we can pull from the quay.io always but for that, we need to update the Rook version in build.env once rook is updated remove changes in the centos branch to pre-pull the ceph image from the internal registry?

> How does Rook refer to the images? If that is the abbreviated name, only then you may need to do the re-tagging with `podman_pull`.

in Rook 1.6.2 which we use in cephcsi now refers to it as `ceph/ceph:v16` https://github.com/rook/rook/blob/v1.6.2/cluster/examples/kubernetes/ceph/cluster-test.yaml#L29 with Rook 1.7 this is referred to as `quay.io/ceph/ceph:v16`.

**Comment by nixpanic** [comment]
If all the required versions are on quay.io, pre-pull them from there, and re-tag them as `ceph/ceph:v16`. The references in `images.txt` can be removed.

We _could_ mirror images locally, that would reduce internet bandwidth, and possibly speed things up a little. But that is not required and can be done in a different PR.

**Comment by Madhu-1** [comment]
closing this one as things are working fine. will revisit when required.

**Pull request closed without merging**